conan/internal/api/install/generators.py

```diff
@@ -151,6 +155,13 @@ def _receive_conf(conanfile):
             conanfile.conf.compose_conf(build_require.conf_info)
 
 
+def _receive_generators(conanfile):
+    """  Collect generators_info from the immediate build_requires"""
+    for build_require in conanfile.dependencies.direct_build.values():
+        if build_require.generator_info:
```

Comment:
We need here the check that this is a list or the conversion to a list, because users will do ``self.generator_info = "MyGenerator"``